### PR TITLE
add test for index in iteration context Refs #12

### PR DESCRIPTION
**Pull request by graingert**
### Description

<describe bug fix or new feature>

Fixes: #

### CODE REVIEW

**Comment by jreese** [comment]
Can you rebase on latest master to get a full test run on CI?

File: fissix/tests/test_fixers.py [hunk 1]
```diff
@@ -3158,6 +3158,10 @@ def test_map_trailers(self):
         a = """x = list(map(f, 'abc'))[0][1]"""
         self.check(b, a)
 
+        b = """x = set(map(f, 'abc')[0])"""
+        b = """x = set(list(map(f, 'abc'))[0])"""
```
Comment (graingert) [inline fissix/tests/test_fixers.py:3162]:
```suggestion
        a = """x = set(list(map(f, 'abc'))[0])"""
```

File: fissix/tests/test_fixers.py [hunk 2]
```diff
@@ -3158,6 +3158,10 @@ def test_map_trailers(self):
         a = """x = list(map(f, 'abc'))[0][1]"""
         self.check(b, a)
 
+        b = """x = set(map(f, 'abc')[0])"""
+        a = """x = set(list(map(f, 'abc'))[0])"""
```
Comment (graingert) [inline fissix/tests/test_fixers.py:3162]: @jreese this now fails as expected, a bug here? https://github.com/jreese/fissix/blob/master/fissix/fixer_util.py#L249-L265

I'm not sure how to fix the pattern though

**Comment by graingert** [comment]
I'm pretty sure #17 won't help, here lib2to3 is handling the detection of iterating context incorrectly, rather than forgetting that dict is one

**Pull request closed without merging**